tests/framework/config/cluster.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+type TLSConfig string
+
+var (
+	NoTLS     TLSConfig = ""
+	AutoTLS   TLSConfig = "auto-tls"
+	ManualTLS TLSConfig = "manual-tls"
```

Comment:
It would be better to define an enum something like below,

```
type TLSConfig int
const (
    NoTLS   TLSConfig = iota
    AutoTLS
    ManualTSL
)
```

Reply:
Do you have any arguments for why it's better to have int then string? Usual reason is performance (cost of storing int vs string), however this enum is just for tests. For tests I would prefer readability vs performance.

Reply:
This is just a nit, feel free to keep it unchanged. .  The proposed code is almost a `standard` way to implementation enum type in golang.

Reply:
How about the mid-ground ?  strings, but with enforced immutability 
```
type TLSConfig int string
const (
	NoTLS     TLSConfig = "no-tls" (or "")
	AutoTLS   TLSConfig = "auto-tls"
	ManualTLS TLSConfig = "manual-tls"
)

Reply:
Yes, `int` or `string` isn't the point. The point is that `const` makes more sense than `var`. A variable's value may be changed accidentally. 

I am asking change for this since it's a nit. It's also OK to change it in a separate PR if it needs a little big effort.

Reply:
Done